worldedit-fabric/src/main/java/com/sk89q/worldedit/fabric/FabricWorldEdit.java

```diff
@@ -205,10 +208,22 @@ private void registerCommands(CommandDispatcher<CommandSourceStack> dispatcher,
     private FabricPermissionsProvider getInitialPermissionsProvider() {
         try {
             Class.forName("me.lucko.fabric.api.permissions.v0.Permissions", false, getClass().getClassLoader());
+            Optional<Version> version = FabricLoader.getInstance().getModContainer("fabric-permissions-api-v0")
+                    .map(ModContainer::getMetadata)
+                    .map(ModMetadata::getVersion);
+
+            if (version.isPresent() && !VersionPredicate.parse(">=0.3.3").test(version.get())) {
+                throw new RuntimeException("Fabric permissions version " + version.get() + " is not supported. Please update Fabric Permissions API");
+            }
+
             return new FabricPermissionsProvider.LuckoFabricPermissionsProvider(platform);
         } catch (ClassNotFoundException ignored) {
             // fallback to vanilla
+        } catch (Throwable e) {
```

Comment:
We should keep this as `Exception` or we risk suppressing real errors.
```suggestion
        } catch (Exception e) {
```

Reply:
My logic here was any error in that mod is something we just want to provide a warn for, rather than actually crash, given we have a fallback available

I'm happy to change it to Exception, but it does partially reduce the intention

Reply:
An `Error` is likely to be OOM or similar fatal exception, so I wouldn't worry about it. Or it could be an `AssertionError` from us passing bad values somewhere, which should be louder I think.